src/libraries/System.ObjectModel/tests/TrimmingTests/TypeConverterAttributeCtorTest.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel;
+using System.Globalization;
+
+namespace TypeConverterAttributeTest
+{
+    /// <summary>
+    /// Tests that the public constructors of types passed into System.ComponentModel.TypeConverterAttribute
+    /// are not trimmed out when needed in a trimmed application.
+    /// </summary>
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            // Type-based TypeConverterAttribute ctor overload, ensure public parameterized ctor of TypeConverter type is preserved.
+            TypeDescriptor.AddAttributes(typeof(DayOfWeek), new TypeConverterAttribute(typeof(EnumConverter)));
+            var attribute = new DefaultValueAttribute(typeof(DayOfWeek), "Friday");
+            if ((DayOfWeek)attribute.Value != DayOfWeek.Friday)
+            {
+                return -1;
+            }
+
+            // String-based TypeConverterAttribute ctor overload, ensure public parameterless ctor of TypeConverter type is preserved.
+            TypeDescriptor.AddAttributes(typeof(string), new TypeConverterAttribute("TypeConverterAttributeTest.MyStringConverter, project, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null"));
```

Comment:
You need to use a different type here to get the test coverage. The Type overload test above would preserve the constructor of this type. And even if we messed up the string overload this test would pass.